src/Mod/TechDraw/Gui/CommandExtensionPack.cpp

```diff
@@ -365,49 +568,1333 @@ bool CmdTechDrawExtensionThreadBoltBottom::isActive(void)
+}
+
+void _intersectionCL(TechDraw::BaseGeom* geom1, TechDraw::BaseGeom* geom2, std::vector<Base::Vector3d>& interPoints){
+    // find intersection vertex(es) between one circle and one line
+    // Taken from: <http://de.wikipedia.org/wiki/Schnittpunkt>
+    TechDraw::Circle* gen1 = static_cast<TechDraw::Circle *>(geom1);
+    TechDraw::Generic* gen2 = static_cast<TechDraw::Generic *>(geom2);
+    Base::Vector3d cirleCenter = gen1->center;
+    Base::Vector3d startPnt = gen2->points.at(0);
+    Base::Vector3d endPnt = gen2->points.at(1);
+    Base::Vector3d dir = endPnt-startPnt;
+    float r0 = gen1->radius;
+    float x0 = cirleCenter.x;
+    float y0 = cirleCenter.y;
+    float a = -dir.y;
+    float b = dir.x;
+    float c = -startPnt.x*dir.y+startPnt.y*dir.x;
+    float d = c-a*x0-b*y0;
+    float ab = a*a+b*b;
+    float rootArg = r0*r0*ab-d*d;
+    if (rootArg > 0){
+        if (rootArg < 0.01){
+            float x1 = x0+a*d/ab;
+            float y1 = -y0+b*d/ab;
+            Base::Vector3d interPoint1(x1,y1,0.0);
+            interPoints.push_back(interPoint1);
+        }
+        else {
+            float root = sqrt(rootArg);
+            float x1 = x0+(a*d+b*root)/ab;
+            float y1 = -y0-(b*d-a*root)/ab;
+            float x2 = x0+(a*d-b*root)/ab;
+            float y2 = -y0-(b*d+a*root)/ab;
+            Base::Vector3d interPoint1(x1,y1,0.0);
+            interPoints.push_back(interPoint1);
+            Base::Vector3d interPoint2(x2,y2,0.0);
+            interPoints.push_back(interPoint2);
+        }
+    }
+}
+
+void _intersectionCC(TechDraw::BaseGeom* geom1, TechDraw::BaseGeom* geom2, std::vector<Base::Vector3d>& interPoints){
+    // find intersection vertex(es) between two circles
+    // Taken from: <http://de.wikipedia.org/wiki/Schnittpunkt>
+    TechDraw::Circle* gen1 = static_cast<TechDraw::Circle *>(geom1);
+    TechDraw::Circle* gen2 = static_cast<TechDraw::Circle *>(geom2);
+    Base::Vector3d Center1 = gen1->center;
+    Base::Vector3d Center2 = gen2->center;
+    float r1 = gen1->radius;
+    float r2 = gen2->radius;
+    float d12 = (Center2-Center1).Length();
+    Base::Vector3d m = (Center2-Center1).Normalize();
+    Base::Vector3d n(-m.y,m.x,0.0);
+    float d0 = (r1*r1-r2*r2+d12*d12)/(2*d12);
+    float rootArg = r1*r1-d0*d0;
+    if (rootArg > 0){
+        if (rootArg < 0.1){
+            Base::Vector3d interPoint1 = -Center1+m*d0;
+            interPoint1.y = -interPoint1.y;
+            interPoints.push_back(interPoint1);
+        }
+        else {
+            float e0 = sqrt(rootArg);
+            Base::Vector3d interPoint1 = Center1+m*d0+n*e0;
+            interPoint1.y = -interPoint1.y;
+            interPoints.push_back(interPoint1);
+            Base::Vector3d interPoint2 = Center1+m*d0-n*e0;
+            interPoint2.y = -interPoint2.y;
+            interPoints.push_back(interPoint2);
+        }
+    }
+}
+
+Base::Vector3d _circleCenter(Base::Vector3d p1, Base::Vector3d p2, Base::Vector3d p3){
```

Comment:
Since the function GetCircleCenter() of the Sketcher module is almost identical it should be moved to a module that TechDraw and Sketcher depend on, e.g. the Part module.

Also note that the function of the Sketcher module received a bug fix about collinear points that is missing here.